src/wasm-type.h

```diff
@@ -576,6 +587,12 @@ struct TypeBuilder {
   // equirecursive mode.
   void setSubType(size_t i, size_t j);
 
+  // Make this type nominal in the sense of the Milestone 4 GC spec, independent
+  // of the current TypeSystem configuration. `supertype` is the index of the
+  // nontrivial supertype, or `NoSupertype` if there is no nontrivial supertype.
+  enum : size_t { NoSupertype = static_cast<size_t>(-1) };
+  void setNominal(size_t i, size_t supertype);
```

Comment:
Can we use `setSubType` (right above) to define subtyping relationships in all modes, instead of adding a new function? I guess we need a new one to mark as "milestone 4" but that could take no parameters.

Reply:
Yes, that seems cleaner. I'll make that change.